apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/authenticators.tsx

```diff
@@ -137,6 +137,24 @@ export const Authenticators: FunctionComponent<AuthenticatorsPropsInterface> = (
             );
         }
 
+        if (authenticator.name === IdentityProviderManagementConstants.BACKUP_CODE_AUTHENTICATOR) {
+            // If there is only one step in the flow, backup code authenticator shouldn't be allowed.
+            if (currentStep === 0) {
+                return false;
+            }
+
+            // If there exist 2FAs in the current step backup code authenticator should be enabled.
+            // Otherwise, it should be disabled.
+            if (SignInMethodUtils.countTwoFactorAuthenticatorsInCurrentStep(
+                currentStep,
+                authenticationSteps
+            ) > 0) {
+                return true;
+            } else {
+                return false;
+            }
```

Comment:
WDYT think about making the code less verbose with just returning the condition, such as,

```suggestion
            return SignInMethodUtils.countTwoFactorAuthenticatorsInCurrentStep(
                currentStep,
                authenticationSteps
            ) > 0;
```